### PR TITLE
Move C2 test to demos, prepare c2 for publishing.

### PR DESCRIPTION
**Pull request by dot-asm**

### CODE REVIEW

**Comment by dot-asm** [comment]
Consider this a "release candidate" for supraseal-c2 crate. For my viewpoint the only thing that is missing is a clarification in c2/README.md about it being available as option in the upstream bellperson. I suppose @vmx could give some colour, like "will be available in release X."

File: demos/c2-test/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[dependencies]
+
+[dev-dependencies]
+supraseal-c2 = "0"
+anyhow = "1.0.26"
+bincode = "1.1.2"
+filecoin-proofs-api = "15.0.0"
+filecoin-proofs-v1 = { package = "filecoin-proofs", version = "15.0.0", default-features = false }
+storage-proofs-core = { version = "15.0.0", default-features = false }
```
Comment (vmx) [inline demos/c2-test/Cargo.toml:16]: Just a note: ideally you depend only on `filecoin-proofs-api` and not the rest. This is our public API, the rest is kind of for internal use.

Though that's not a blocker and I'm happy to provide a PR myself cleaning things up, once we have a release of `supraseal-c2`.

**Review by vmx — COMMENTED** [review]
It will be available in bellperson v0.26 when you enable the `cuda-supraseal` feature flag.

File: demos/c2-test/Cargo.toml (hunk 1 above)
Comment (dot-asm) [reply demos/c2-test/Cargo.toml:16]: Thanks! (And I've updated the README.md:-)

**Comment by dot-asm** [comment]
Merged, crate is published.

**Pull request closed without merging**